src/main/java/org/hl7/davinci/ehrserver/interceptor/OrderIdentifierAdditionInterceptor.java

```diff
@@ -0,0 +1,97 @@
+import org.hl7.davinci.ehrserver.interceptor.impl.ServiceRequestOrderImpl;
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.*;
+import java.util.stream.Collectors;
+import java.util.List;
+import java.util.UUID;
+
+@Interceptor
+public class OrderIdentifierAdditionInterceptor {
+
+    static final Logger logger = LoggerFactory.getLogger(OrderIdentifierAdditionInterceptor.class.getName());
+
+    private static final String CODE_PLAC = "PLAC";
+
+    public OrderIdentifierAdditionInterceptor() {
+
+    }
+
+    @Hook(Pointcut.STORAGE_PRESTORAGE_RESOURCE_CREATED)
+    public void insert(IBaseResource theResource) {
+        switch (theResource.fhirType()) {
+            case "DeviceRequest":
+            case "ServiceRequest":
+            case "MedicationRequest":
+            case "MedicationDispense":
+                processIdentifier(theResource);
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void processIdentifier(IBaseResource theResource) {
+        logger.info("Processing order identifier");
+        switch (theResource.fhirType()) {
+            case "DeviceRequest":
+                DeviceRequest deviceRequest = ((DeviceRequest) theResource);
+                DeviceRequestOrderImpl deviceRequestImpl = new DeviceRequestOrderImpl(deviceRequest);
+                addIdentifier(deviceRequestImpl);
+                deviceRequest = (DeviceRequest) deviceRequestImpl.getRequest();
+                break;
+            case "ServiceRequest":
+                ServiceRequest serviceRequest = (ServiceRequest) theResource;
+                ServiceRequestOrderImpl serviceRequestOrderImpl = new ServiceRequestOrderImpl(serviceRequest);
+                addIdentifier(serviceRequestOrderImpl);
+                serviceRequest = (ServiceRequest) serviceRequestOrderImpl.getRequest();
+                break;
+            case "MedicationRequest":
+                MedicationRequest medicationRequest = (MedicationRequest) theResource;
+                MedicationRequestOrderImpl medicationRequestOrderImpl = new MedicationRequestOrderImpl(
+                        medicationRequest);
+                addIdentifier(medicationRequestOrderImpl);
+                medicationRequest = (MedicationRequest) medicationRequestOrderImpl.getRequest();
+                break;
+            case "MedicationDispense":
+                MedicationDispense medicationDispense = (MedicationDispense) theResource;
+                MedicationDispenseOrderImpl medicationDispenseOrderImpl = new MedicationDispenseOrderImpl(
+                        medicationDispense);
+                addIdentifier(medicationDispenseOrderImpl);
+                medicationDispense = (MedicationDispense) medicationDispenseOrderImpl.getRequest();
+                break;
+            default:
+                break;
+        }
+    }
+
+    private <T extends OrderInterface> void addIdentifier(T request) {
```

Comment:
Rather than passing this a request, which can variably be one of the four request classes, we can instead pass it the information it needs from the request which is identical between requests.  For example, the arguments could be `Identifier identifier, String ResourceType` to replace `request.getIdentifier()` and `request.getResourceType()`, then instead of being `void` we can return an `Identifier`, which gets added to the request outside of the function in the switch/case above this.